data/store/mongo/mongo_datum.go

```diff
@@ -667,13 +628,18 @@ func (d *DatumRepository) getTimeRange(ctx context.Context, userId string, typ s
 	selector := bson.M{
 		"_active": true,
 		"_userId": userId,
-		"type":    typ,
 		"time": bson.M{
 			"$gte": pastCutoff,
 			"$lte": futureCutoff,
 		},
 	}
 
+	if len(typ) > 1 {
```

Comment:
The special case here should be `len(typ) == 1`, because `$in` can handle empty arrays. Currently, there is a bug when `typ` is `nil` or `empty`. In addition to adding a guard for the length I think you should flip the logic.

Reply:
done